File: src/main/java/com/freya02/botcommands/annotations/api/annotations/CommandId.java
package com.freya02.botcommands.annotations.api.annotations;

import com.freya02.botcommands.api.application.CommandPath;
import com.freya02.botcommands.api.application.GuildApplicationSettings;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Allows you to set a command ID to your application command methods
 *
 * See {@link GuildApplicationSettings#getGuildsForCommandId(String, CommandPath)} for more details
 *
 * @see GuildApplicationSettings#getGuildsForCommandId(String, CommandPath)
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface CommandId {
	/**
	 * @return The ID of this command method, must be unique
	 */
	String value();
}

File: BotCommands-commands/app/src/main/kotlin/io/github/freya022/botcommands/api/commands/application/annotations/CommandId.kt
package io.github.freya022.botcommands.api.commands.application.annotations

/**
 * Sets a **unique** command ID on an **annotated** application command function.
 *
 * This can be used to disambiguate some commands,
 * and can be applied to any application command, top-level or not.
 *
 * @see DeclarationFilter @DeclarationFilter
 */
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class CommandId(val value: String)
